exporters/otlp/otlpmetric/internal/oconf/envconfig.go

```diff
@@ -148,3 +151,35 @@ func withTLSConfig(c *tls.Config, fn func(*tls.Config)) func(e *envconfig.EnvOpt
 		}
 	}
 }
+
+// WithEnvTemporalitySelector sets the temporality selector based on the environment variable.
+// See https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/metrics/sdk_exporters/otlp.md#additional-configuration
+// for more details.
+func WithEnvTemporalitySelector(n string, fn func(metric.TemporalitySelector)) func(e *envconfig.EnvOptionsReader) {
+	return func(e *envconfig.EnvOptionsReader) {
+		if v, ok := e.GetEnvValue(n); ok {
+			switch strings.ToLower(v) {
+			case "delta":
+				fn(deltaTemporality)
+			case "lowmemory":
+				fn(lowMemoryTemporality)
+			case "cumulative":
+				fn(cumulativeTemporality)
+			}
+		}
+	}
+}
+
+func deltaTemporality(ik metric.InstrumentKind) metricdata.Temporality {
+	return metricdata.DeltaTemporality
+}
+func cumulativeTemporality(ik metric.InstrumentKind) metricdata.Temporality {
+	return metricdata.CumulativeTemporality
+}
+func lowMemoryTemporality(ik metric.InstrumentKind) metricdata.Temporality {
+	switch ik {
+	case metric.InstrumentKindCounter, metric.InstrumentKindHistogram:
+		return metricdata.DeltaTemporality
+	}
+	return metricdata.CumulativeTemporality
+}
```

Comment:
nit: Can we add blank lines between the functions?

Side note: Have we ever considered using `gofumpt` linter instead of `gofmt`? 😉